addons/website_livechat/static/src/models/thread/thread.js

```diff
@@ -0,0 +1,40 @@
+odoo.define('website_livechat/static/src/models/thread/thread.js', function (require) {
+'use strict';
+
+require('im_livechat/static/src/models/thread/thread.js'); // ensure this override is loaded first
+
+const {
+    registerClassPatchModel,
+    registerFieldPatchModel,
+} = require('mail/static/src/model/model_core.js');
+const { many2one } = require('mail/static/src/model/model_field.js');
+
+
+registerClassPatchModel('mail.thread', 'website_livechat/static/src/models/thread/thread.js', {
+
+    //----------------------------------------------------------------------
+    // Public
+    //----------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    convertData(data) {
+        const data2 = this._super(data);
+        if ('visitor' in data && data.visitor && !data.livechat_visitor.id) {
```

Comment:
was is the issue with `livechat_visitor` here?

Reply:
If the livechat visitor has an id, then it's not a website visitor, it's a normal user

Reply:
If that's the case, more should be done, at least calling `unlink` with an "else" + add comment to explain.